src/factories/ethers-interface.ts

```diff
@@ -32,3 +28,37 @@ export async function ethersInterfaceFromSpec(spec: FakeContractSpec): Promise<e
     return new ethers.utils.Interface(foundInterface);
   }
 }
+
+async function ethersInterfaceFromAbi(abi: string): Promise<ethers.utils.Interface> {
+  try {
+    return new ethers.utils.Interface(abi);
+  } catch (err) {
+    const error: Error = err as Error;
+    throw new Error(`unable to generate smock spec from abi string, ${error.message}`);
+  }
+}
+
+async function ethersInterfaceFromContractName(contractNameOrFullyQualifiedName: string): Promise<ethers.utils.Interface> {
+  let error: Error | null = null;
+  try {
+    return (await (hre as any).ethers.getContractFactory(contractNameOrFullyQualifiedName)).interface;
+  } catch (err) {
+    error = err as Error;
+  }
+
+  try {
+    return (await (hre as any).ethers.getContractAt(contractNameOrFullyQualifiedName, ethers.constants.AddressZero)).interface;
+  } catch (err) {
+    error = err as Error;
+  }
+
+  throw new Error(`unable to generate smock spec from contract name, ${error.message}`);
+}
+
+function isJson(str: string): boolean {
```

Comment:
This is a minimal identifier function that can determine the possibility of the input string / spec to potentially be json string or not.  I've not used `JSON.parse(str)` here because we don't care if the string is a valid json.

Reply:
Could we just do:

```
    try {
        JSON.parse(str)
    } catch (e) {
        return false
    }
    return true
```

Instead? Otherwise I would recommend changing the name of the function to `isMaybeJson`

Reply:
The problem with using `JSON.parse()` for the minimal json test is that it would return a `false` if the ABI json string was malformed / invalid json. But in this scenario we actually need to identify that the input was in-fact a json string and then delegate to the json string input handler function - `ethersInterfaceFromAbi`.
As advised, I'll update the function name to better convey the intention.